packages/flutter/lib/src/widgets/app.dart

```diff
@@ -236,6 +236,9 @@ typedef InitialRouteListFactory = List<Route<dynamic>> Function(String initialRo
 /// It is used by both [MaterialApp] and [CupertinoApp] to implement base
 /// functionality for an app.
 ///
+/// If a [MediaQuery] is not available above [WidgetsApp], a [MediaQuery] is
```

Comment:
So, this is very very breaking after checking out all the affected testes internally.  I think this should be made an optional configuration, that way folks can opt-in to this behavior if they want it. @goderbauer was one of the original reviewers, WDYT?

Reply:
Yeah, if this is too breaking we need to find a way to make it less breaking.

Reply:
Can I help somehow @Piinks?

Reply:
Yes! Can you update this to make this an opt-in feature? That way folks won't be broken, but those who want to use it can turn it on. :)

Reply:
Done @Piinks! ✅
Let me know if it is alright